src/contracts/atlas/Atlas.sol

```diff
@@ -221,4 +220,41 @@ contract Atlas is Test, Factory {
         } 
         return IExecutionEnvironment(executionEnvironment).validateUserCall(userMetaTx);
     }
+
+    function metacallSimulation(
+        ProtocolCall calldata protocolCall,
+        UserCall calldata userCall,
+        SearcherCall[] calldata searcherCalls,
+        Verification calldata verification
+    ) external payable {
+        if (!metacall(protocolCall, userCall, searcherCalls, verification)) {
+            revert("ERR-S01 NoAuctionWinner");
```

Comment:
We could use custom errors here and look for their bytes4 selector in the catch

Reply:
Done in https://github.com/Polygon-Fast-Lane/atlas/pull/15/commits/d82f00d81f8ec88a26548adcaeb2574f7be03c84